lib/rouge/lexers/cpp.rb

```diff
@@ -77,6 +78,12 @@ def self.reserved
         rule %r/[.]{3}/, Operator
         mixin :whitespace
       end
+
+      state :template do
+        rule %r/>/, Punctuation, :pop!
+        rule %r/(class|struct|typename)\b/, Keyword, :classname
```

Comment:
not sure if I know enough to comment on anything here, but if I interpreted this correctly, I'd say there's no need to check for `struct` in a template declaration, as only `class` and `typename` are valid.

Reply:
yeah actually looking at the tests below, I think I see this is not only for the contents within the angle bracket `<>` pair, so discard what I said

Reply:
Actually, since I `mixin` the `:root` state later I don't think I really need either `class` or `struct` here. I'll tidy that up.